concordium-consensus/src/Concordium/GRPC2.hs

```diff
@@ -535,11 +563,714 @@ instance ToProto AccountInfo where
+                                ProtoFields.sender .= toProto sender'
+                                ProtoFields.amount .= toProto etAmount
+                                ProtoFields.receiver .= toProto receiver)
+                        Interrupted{..} -> Right . Proto.make $ ProtoFields.interrupted .= Proto.make (do
+                            ProtoFields.address .= toProto iAddress
+                            ProtoFields.events .= map toProto iEvents)
+                        Resumed{..} -> Right . Proto.make $ ProtoFields.resumed .= Proto.make (do
+                            ProtoFields.address .= toProto rAddress
+                            ProtoFields.success .= rSuccess)
+                        _ -> Left CEInvalidTransactionResult
+                v <- mapM toContractEvent events
+                Right . Proto.make $ ProtoFields.contractUpdateIssued . ProtoFields.effects .= v
+            TTTransfer-> mkSuccess <$> do
+                v <- case events of
+                    [Transferred{..}] -> case etTo of
+                        AddressContract _ -> Left CEInvalidTransactionResult
+                        AddressAccount receiver -> Right . Proto.make $ do
+                            ProtoFields.amount .= toProto etAmount
+                            ProtoFields.receiver .= toProto receiver
+                    _ -> Left CEInvalidTransactionResult
+                Right . Proto.make $ ProtoFields.accountTransfer .= v
+            TTTransferWithMemo -> mkSuccess <$> do
+                v <- case events of
+                    [Transferred{..}, TransferMemo{..}] -> case etTo of
+                        AddressContract _ -> Left CEInvalidTransactionResult
+                        AddressAccount receiver -> Right . Proto.make $ do
+                            ProtoFields.amount .= toProto etAmount
+                            ProtoFields.receiver .= toProto receiver
+                            ProtoFields.memo .= toProto tmMemo
+                    _ -> Left CEInvalidTransactionResult
+                Right . Proto.make $ ProtoFields.accountTransfer .= v
+            TTAddBaker -> mkSuccess <$> do
+                v <- case events of
+                    [BakerAdded{..}] -> Right $ toProto ((ebaBakerId, ebaAccount, ebaSignKey, ebaElectionKey, ebaAggregationKey), ebaStake, ebaRestakeEarnings)
+                    _ -> Left CEInvalidTransactionResult
+                Right . Proto.make $ ProtoFields.bakerAdded .= v
+            TTRemoveBaker -> mkSuccess <$> do
+                v <- case events of
+                    [BakerRemoved{..}] -> Right $ toProto ebrBakerId
+                    _ -> Left CEInvalidTransactionResult
+                Right . Proto.make $ ProtoFields.bakerRemoved .= v
+            TTUpdateBakerStake -> mkSuccess <$> do
+                v <- case events of
+                    [] -> Right Proto.defMessage -- TODO: Review question: Is it correct that if the stake is unchanged, no events occur?
```

Comment:
Yes, it is unfortunately correct  and requires this special case. See line 1845 in the Scheduler.